addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -263,12 +263,22 @@ async function applyModifications(img, dataOptions = {}) {
  * @param {String} src URL of the image to load
  * @param {HTMLImageElement} [img] img element in which to load the image
  * @returns {Promise<HTMLImageElement>} Promise that resolves to the loaded img
+ *     or a placeholder image if the src is not found.
  */
 function loadImage(src, img = new Image()) {
     return new Promise((resolve, reject) => {
-        img.addEventListener('load', () => resolve(img), {once: true});
-        img.addEventListener('error', reject, {once: true});
-        img.src = src;
+        fetch(src)
+            .then(response => {
+                if (response.status === 404) {
+                    // The server will return a placeholder image with the
+                    // following src.
+                    src = "/web/image/__odoo__unknown__src__/";
+                }
+                img.addEventListener('load', () => resolve(img), {once: true});
+                img.addEventListener('error', reject, {once: true});
+                img.src = src;
+            })
+            .catch(reject);
```

Comment:
Shouldn't we also handle the catch (could be a cors protected absolute image from external domain (or other domain accessing the same db)) as a 404?

Reply:
Good remark. I also changed the code so that any not successful response will have the fallback image :+1: